### ISSUE TITLE
Hover on local shows information in function call position

### ISSUE THREAD

**Issue by snoe** [clojure-lsp/clojure-lsp#1281]
**Describe the bug**
Hover is showing information for things not under the cursor. Even though parameters and local bindings have little information to show it's better to be told it's a local than showing information for something unrelated that could (in a long function) be completely off screen.

**To Reproduce**
Given the following code and the cursor `|` at these three positions (p1,p2,p3).
```clojure
(def d {})
(defn foo [|a & _]
  (|a)
  (map |a d))
```
At p2 hover shows `a` as expected.

However:
At p1 hover shows the information for `defn`, I would expect `a`
At p3 hover shows the information for `map`, I would expect `a`

**User details (please complete the following information):**
master (f6481e790c9e35105bec6328af82234cbacbdb9d)
and
2022.09.01-15.27.31

**Comment by ericdallo** [comment]
@snoe that was actually intended as `a` is a local, when coding is useful to see the hover docs of the function being called otherwise the current symbol, for example:
```clojure
(foo |a b c)
```

It's useful to see foo docs to check arity and how many args the function has, unless a, b or c are another function call or something, do you have a more concrete example of why that would be better?

**Comment by mainej** [comment]
I’ve never quite understood the difference between `hover` and `signatureHelp`. Is that relevant here?

**Comment by ericdallo** [comment]
signatureHelp was intended to help with that, but most editors and users rely on hover to see function docs/arity/examples, emacs for example, shows the current function on eldoc while typing.

**Comment by snoe** [comment]
@ericdallo it feels really inconsistent and confusing and afaict goes against the purpose of hover.
1. inconsistent:`(map |a d)` if a is a local or var, you get different behavior, so hover to get signature only works sometimes and from a user perspective I think it would be hard to know whether it will or not.
2. crosspurpose: I often lose track of what's a var or a local, hover should be able to tell me that and deducing that "oh I saw the call hover, so that must be a local" is janky.
3. confusing: github image is upload isn't working, so I can't show real world but it's fairly common that hover on a local and I get docs for something that's completely offscreen. In this instance imagine the `if` is above the fold, and you hover on z, it feels very wrong to get a hover for the if.
```clojure
(if x
  {:a a
   :b b
   :c c
   :d d
   ... ...
   :z |z}
   ...)
```

**Comment by ericdallo** [comment]
The idea for that was: why we would want to show locals docs if it's a local? We always try to show the current function being called docs

**Comment by ericdallo** [comment]
Yeah, I can see how the 3 example is weird, but showing to me `z` on hover doesn't help either. I really think showing hover for 
`map` in `(map |a b)` helps and it'd be nice to keep supporting that...
Do you think we should support both somehow? the code would not be hard to maintain, I know signatureHelp is a feature more intended to show that, but I doubt we would manage to change how clients present hover or signature help

**Comment by snoe** [comment]
I think maybe you're approaching it as hover == docs; af `a` was a function var, you would expect the docs for `a` to show up right? I'm approaching it as hover is about what's under my cursor, I think that could include surrounding context like the surrounding function name and signature. 

So for consistency, hover could:
- in call position do what it does now
- in arg position show:

> **calling:** `clojure.core/map [f] [f coll] [f c1 c2] [f c1 c2 c3] [f c1 c2 c3 & colls]`
> ----
> <hover information about what's under cursor>

e.g. **below the hr** for `(map |a b)` it would just show `a`, of `(def a inc) (map |a b)` it would show `user/a`, for `(map |inc b)` it would show the `inc` docs.

**Comment by ericdallo** [comment]
I see, sounds good to me, especially if we keep showing call as the first line

**Comment by TheBlob42** [comment]
> I think maybe you're approaching it as hover == docs; if `a` was a function var, you would expect the docs for `a` to show up right? I'm approaching it as hover is about what's under my cursor, I think that could include surrounding context like the surrounding function name and signature.
> 
> So for consistency, hover could:
> 
>     * in call position do what it does now
> 
>     * in arg position show:
> 
> 
> > ## **calling:** `clojure.core/map [f] [f coll] [f c1 c2] [f c1 c2 c3] [f c1 c2 c3 & colls]`
> > <hover information about what's under cursor>
> 
> e.g. **below the hr** for `(map |a b)` it would just show `a`, of `(def a inc) (map |a b)` it would show `user/a`, for `(map |inc b)` it would show the `inc` docs.

I really like this proposal and therefore created a PR for it here: #1727 

Please have a look into the code and try out the functionality if this is how you would imagine it :slightly_smiling_face:

**Comment by ericdallo** [comment]
Fixed by #1733 1735
